File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/viewmodel/item/ItemBannerChildVModel.java
package io.ganguo.demo.viewmodel.item;

import android.databinding.ObservableField;
import android.view.View;

import io.ganguo.demo.R;
import io.ganguo.demo.databinding.ItemBannerChildBinding;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.view.AdapterInterface;

/**
 * Created by leo on 2017/4/27.
 * 广告轮播 - ItemViewModel
 */
public class ItemBannerChildVModel extends BaseViewModel<AdapterInterface<ItemBannerChildBinding>> {
    private ObservableField<String> imageUrl = new ObservableField<>();

    @Override
    public int getItemLayoutId() {
        return R.layout.item_banner_child;
    }

    @Override
    public void onViewAttached(View view) {

    }

    public ItemBannerChildVModel setImageUrl(String imageUrl) {
        this.imageUrl.set(imageUrl);
        return this;
    }

    public ObservableField<String> getImageUrl() {
        return imageUrl;
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/viewmodel/item/ItemBannerChildVModel.kt
package io.ganguo.demo.viewmodel.item

import androidx.databinding.ObservableField
import android.view.View

import io.ganguo.demo.R
import io.ganguo.demo.databinding.ItemBannerChildBinding
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.core.viewinterface.AdapterInterface

/**
 * Created by leo on 2017/4/27.
 * 广告轮播 - ItemViewModel
 */
class ItemBannerChildVModel : BaseViewModel<io.ganguo.viewmodel.core.viewinterface.AdapterInterface<ItemBannerChildBinding>>() {
    val imageUrl = ObservableField<String>()
    override val layoutId: Int by lazy { R.layout.item_banner_child }
    override fun onViewAttached(view: View) {

    }

    fun setImageUrl(imageUrl: String): ItemBannerChildVModel {
        this.imageUrl.set(imageUrl)
        return this
    }

}
